pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -0,0 +1,368 @@
+		CanFail: canConditionalGathererFail,
+	}
+
+	for _, conditionalGathering := range gatheringRules {
+		allConditionsAreSatisfied, err := g.areAllConditionsSatisfied(conditionalGathering.Conditions)
+		if err != nil {
+			return nil, err
+		}
+
+		if allConditionsAreSatisfied {
+			functions, errs := g.createGatheringClosures(conditionalGathering.GatheringFunctions)
+			if len(errs) > 0 {
+				return nil, err
+			}
+
+			for funcName, function := range functions {
+				gatheringFunctions[funcName] = function
+			}
+		}
+	}
+
+	return gatheringFunctions, nil
+}
+
+// GatherConditionalGathererRules stores the gathering rules in insights-operator/conditional-gatherer-rules.json
+func (g *Gatherer) GatherConditionalGathererRules(context.Context) ([]record.Record, []error) {
+	return []record.Record{
+		{
+			Name: "insights-operator/conditional-gatherer-rules",
+			Item: record.JSONMarshaller{Object: gatheringRules},
+		},
+	}, nil
+}
+
+// areAllConditionsSatisfied returns true if all the conditions are satisfied, for example if the condition is
+// to check if a metric is firing, it will look at that metric and return the result according to that
+func (g *Gatherer) areAllConditionsSatisfied(conditions []conditionWithParams) (bool, error) {
+	for _, condition := range conditions {
+		switch condition.Type {
+		case alertIsFiring:
+			alertName, err := getStringFromMap(condition.Params, "name")
+			if err != nil {
+				return false, err
+			}
+
+			if !g.isAlertFiring(alertName) {
+				return false, nil
+			}
+		default:
+			return false, fmt.Errorf("unknown condition type: %v", condition.Type)
+		}
+	}
+
+	return true, nil
+}
+
+// updateAlertsCache updates the cache with firing alerts
+func (g *Gatherer) updateAlertsCache(ctx context.Context) error {
+	if g.metricsGatherKubeConfig == nil {
+		return nil
+	}
+
+	metricsClient, err := rest.RESTClientFor(g.metricsGatherKubeConfig)
+	if err != nil {
+		return err
+	}
+
+	return g.updateAlertsCacheFromClient(ctx, metricsClient)
+}
+
+func (g *Gatherer) updateAlertsCacheFromClient(ctx context.Context, metricsClient rest.Interface) error {
+	g.firingAlerts = make(map[string]bool)
+
+	data, err := metricsClient.Get().AbsPath("federate").
+		Param("match[]", `ALERTS{alertstate="firing"}`).
+		DoRaw(ctx)
+	if err != nil {
+		return err
+	}
+
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(bytes.NewReader(data))
+	if err != nil {
+		return err
+	}
+
+	if len(metricFamilies) > 1 {
+		// just log cuz everything would still work
+		klog.Warning("unexpected output from prometheus metrics parser")
+	}
+
+	metricFamily, found := metricFamilies["ALERTS"]
+	if !found {
+		klog.Info("no alerts are firing")
+		return nil
+	}
+
+	for _, metric := range metricFamily.GetMetric() {
+		if metric == nil {
+			klog.Info("metric is nil")
+			continue
+		}
+
+		for _, label := range metric.GetLabel() {
+			if label == nil {
+				klog.Info("label is nil")
```

Comment:
I guess seeing this message (and similar above) in the IO log will not be very helpful :) on the other hand I am not having any better suggestion at hand (maybe saying something that the metric family with given name references some empty metrics or something like that). It's not blocker for merging.